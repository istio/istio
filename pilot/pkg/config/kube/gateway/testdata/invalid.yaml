apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: invalid-service
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.example"
    port: 80
    protocol: HTTP
  addresses:
  - value: fake-service.com
    type: Hostname
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: target-port-reference
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.example"
    port: 8080 # Test service has port 80 with targetPort 8080
    protocol: HTTP
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: invalid-gateway-address
  namespace: invalid-gateway-address
spec:
  gatewayClassName: istio
  addresses:
  - value: 1.2.3.4
    type: istio.io/FakeType
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: invalid-cert-kind
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "domain.example"
    port: 34000
    protocol: HTTPS
    tls:
      mode: Terminate
      certificateRefs:
      - name: my-cert-http
        group: core
        kind: unknown
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: invalid-cert-notfound
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "domain.example"
    port: 34001
    protocol: HTTPS
    tls:
      mode: Terminate
      certificateRefs:
      - name: nonexistent
        kind: Secret
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: invalid-cert-malformed
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "domain.example"
    port: 34002
    protocol: HTTPS
    tls:
      mode: Terminate
      certificateRefs:
      - name: malformed
        kind: Secret
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: udp-protocol
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
    - name: udp
      port: 1234
      protocol: UDP
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: unknown-protocol
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
    - name: unknown
      port: 1234
      protocol: unknown
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: protocol-lower-case
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
    - name: http
      port: 1234
      protocol: http
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-backendRef-kind
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["first.domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      kind: GcsBucket
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-backendRef-notfound
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["second.domain.example"]
  rules:
  - backendRefs:
    - name: nonexistent
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-backendRef-mixed
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["third.domain.example"]
  rules:
  - backendRefs:
    - name: nonexistent
      port: 80
      weight: 1
    - name: httpbin
      port: 80
      weight: 1
    - name: httpbin
      kind: GcsBucket
      weight: 1
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-mirror
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - filters:
    - type: RequestMirror
      requestMirror:
        backendRef:
          kind: no-support
          name: httpbin-mirror
          port: 80
    backendRefs:
    - name: httpbin
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: no-backend
  namespace: default
spec:
  parentRefs:
  - group: ""
    kind: Service
    name: httpbin
  rules:
  - filters:
    - type: RequestMirror
      requestMirror:
        backendRef:
          name: httpbin
          port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-parentRef-port
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    port: 1234
  hostnames: ["first.domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
      weight: 1
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: invalid-sectionname-port
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: fake
  hostnames: ["first.domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
      weight: 1
