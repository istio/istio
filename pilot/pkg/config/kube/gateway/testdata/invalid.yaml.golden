apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: fake-service.com
    internal.istio.io/parents: Gateway/invalid-service/default.istio-system
  creationTimestamp: null
  name: invalid-service-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/*.example
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/target-port-reference/default.istio-system
  creationTimestamp: null
  name: target-port-reference-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/*.example
    port:
      name: default
      number: 8080
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-backendRef-hostname.default,HTTPRoute/invalid-backendRef-mirror.default,HTTPRoute/invalid-backendRef-serviceimport.default,HTTPRoute/no-backend.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: default~httpbin.default.svc.domain.suffix
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - httpbin.default.svc.domain.suffix
  http:
  - name: default.invalid-backendRef-hostname.0
    route:
    - destination:
        host: unknown.example.com
        port:
          number: 80
  - name: default.invalid-backendRef-mirror.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - name: default.invalid-backendRef-serviceimport.0
    route:
    - destination:
        host: unknown-service-import.default.svc.domain.suffix
        port:
          number: 80
  - directResponse:
      status: 500
    mirrors:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
    name: default.no-backend.0
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-parentRef-service.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: default~not-found.default.svc.domain.suffix
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - not-found.default.svc.domain.suffix
  http:
  - name: default.invalid-parentRef-service.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-mirror.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~*
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  http:
  - name: default.invalid-mirror.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-backendRef-kind.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~first.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - first.domain.example
  http:
  - name: default.invalid-backendRef-kind.0
    route:
    - destination: {}
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-backendRef-notfound.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~second.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - second.domain.example
  http:
  - name: default.invalid-backendRef-notfound.0
    route:
    - destination:
        host: nonexistent.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/invalid-backendRef-mixed.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~third.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - third.domain.example
  http:
  - name: default.invalid-backendRef-mixed.0
    route:
    - destination:
        host: nonexistent.default.svc.domain.suffix
        port:
          number: 80
      weight: 1
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
      weight: 1
    - destination: {}
      weight: 1
---
